### PR TITLE
Spidercoordinator: It able to get CIDR from kubeadm-config

### PR DESCRIPTION
**Pull request by cyclinder**
If the kube-controller-manager Pod is running as systemd precess rather than Pod, In this case, We can't get the CIDR from the KCM Pod. We can get the CIDR from the kubeadm-config configMap.

#### What type of PR is this?

**What this PR does / why we need it**:

Cherry-pick

**Which issue(s) this PR fixes**:

Fixes https://github.com/spidernet-io/spiderpool/issues/3126

**Special notes for your reviewer**:

### CODE REVIEW

File: test/e2e/spidercoordinator/spidercoordinator_test.go
Comment (ty-dc) [inline test/e2e/spidercoordinator/spidercoordinator_test.go:377]: 377 行此处格式有误。 `}()` 错误的用法，正确的应该是 `})`, 并且它还缺少 })。格式应该如下？

```
	Context("It can get the clusterCIDR from kubeadmConfig and kube-controller-manager pod", Label("V00009"), func() {

		var spc *spiderpoolv2beta1.SpiderCoordinator
		var cm *corev1.ConfigMap
		var err error
		BeforeEach(func() {
			if !common.CheckRunOverlayCNI() {
				GinkgoWriter.Println("This environment is in underlay mode.")
				Skip("Not applicable to underlay mode")
			}

			if !common.CheckCalicoFeatureOn() {
				GinkgoWriter.Println("The CNI isn't calico.")
				Skip("This case only run in calico")
			}

			cm, err = frame.GetConfigmap("kubeadm-config", "kube-system")
			Expect(err).NotTo(HaveOccurred())

			spc, err = GetSpiderCoordinator(common.SpidercoodinatorDefaultName)
			Expect(err).NotTo(HaveOccurred(), "failed to get SpiderCoordinator, error is %v", err)

			// Switch podCIDRType to `cluster`.
			spcCopy := spc.DeepCopy()
			spcCopy.Spec.PodCIDRType = pointer.String(common.PodCIDRTypeCluster)
			Expect(PatchSpiderCoordinator(spcCopy, spc)).NotTo(HaveOccurred())

			DeferCleanup(func() {
				spc, err := GetSpiderCoordinator(common.SpidercoodinatorDefaultName)
				Expect(err).NotTo(HaveOccurred(), "failed to get SpiderCoordinator, error is %v", err)
				GinkgoWriter.Printf("Display the default spider coordinator information: %+v \n", spc)

				// Switch podCIDRType to `auto`.
				spcCopy := spc.DeepCopy()
				spcCopy.Spec.PodCIDRType = pointer.String(common.PodCIDRTypeAuto)
				Expect(PatchSpiderCoordinator(spcCopy, spc)).NotTo(HaveOccurred())

				Eventually(func() bool {
					spc, err := GetSpiderCoordinator(common.SpidercoodinatorDefaultName)
					Expect(err).NotTo(HaveOccurred(), "failed to get SpiderCoordinator, error is %v", err)
					GinkgoWriter.Printf("Display the default spider coordinator information: %+v \n", spc)

					if spc.Status.OverlayPodCIDR == nil || spc.Status.Phase != coordinatormanager.Synced {
						GinkgoWriter.Printf("status.overlayPodCIDR status is still synchronizing, status %+v \n", spc.Status.OverlayPodCIDR)
						return false
					}

					for _, cidr := range spc.Status.OverlayPodCIDR {
						if ip.IsIPv4CIDR(cidr) {
							if cidr != v4PodCIDRString {
								return false
							}
							GinkgoWriter.Printf("ipv4 podCIDR is as expected, value %v=%v \n", cidr, v4PodCIDRString)
						} else {
							if cidr != v6PodCIDRString {
								return false
							}
							GinkgoWriter.Printf("ipv6 podCIDR is as expected, value %v=%v \n", cidr, v6PodCIDRString)
						}
					}
					return true
				}, common.ExecCommandTimeout, common.ForcedWaitingTime).Should(BeTrue())
			})
		})
```

Comment (cyclinder) [reply test/e2e/spidercoordinator/spidercoordinator_test.go]: 怎么自动cherry-pick 成这样了..

**Comment by ty-dc** [comment]
CI 失败的原因：

release-0.8 的 e2e 安装的 IPv6 单栈集群的 calico 有 bug，它仍有 IPV4、IPv6 的池。故在 IPv6 的集群中 auto 模式下仍获取到了 ipv4 的 pod CIDR。

将 release-0.8 的 test/scripts/install-default-cni.sh 中 151 行修改一下？保证 ipv6 模式下只有calico 的 ipv6 池？
```bash
      ipv6)
        kubectl delete ippools default-ipv4-ippool --force
        kubectl patch ippools default-ipv6-ippool --patch '{"spec": {"cidr": "'"${CALICO_IPV6POOL_CIDR}"'"}}' --type=merge
        ;;
```

**Comment by weizhoublue** [comment]
[FAIL] SpiderCoordinator Switch podCIDRType to `calico` or `cilium`、`none`

https://github.com/spidernet-io/spiderpool/actions/runs/7638721733/job/20818914107?pr=3130

**Review by ty-dc — APPROVED** [review]

**Review by weizhoublue — APPROVED** [review]

**Comment by cyclinder** [comment]
why can't merge this when we have two approves?

**Comment by cyclinder** [comment]
> The base branch requires all commits to be signed

**Pull request merged**
